### PR TITLE
bridge/opentracing: set span event name from the event log field

### LINKED ISSUE #8643: bridge/opentracing: LogFields/LogKV always emit AddEvent with an empty name, violating OTLP's "semantically required" non-empty Event.name
### Description

`bridgeSpan.LogFields` in `bridge/opentracing/bridge.go` translates every
OpenTracing `Span.LogFields`/`Span.LogKV` call into an OTel span event with a
**hardcoded empty name**:

```go
func (s *bridgeSpan) LogFields(fields ...otlog.Field) {
	s.otelSpan.AddEvent(
		"",
		trace.WithAttributes(otLogFieldsToOTelAttrs(fields)...),
	)
}
```
https://github.com/open-telemetry/opentelemetry-go/blob/main/bridge/opentracing/bridge.go#L157-L162

The OpenTracing API has no explicit "event name" parameter on `LogFields`,
but by convention a field keyed `"event"` is meant to carry the event's
name/description (see the OpenTracing spec / `opentracing-go/log`
package docs). This bridge ignores that convention entirely and always
passes `""` as the OTel event name, regardless of what fields are logged.

This becomes a real interoperability bug downstream: the OTLP spec's
`Span.Event.name` field is documented as **"semantically required to be set
to non-empty string"**:

https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/trace/v1/trace.proto#L220-L228
```protobuf
message Event {
  fixed64 time_unix_nano = 1;

  // The name of the event.
  // This field is semantically required to be set to non-empty string.
  string name = 2;
  ...
}
```

Backends that enforce this (e.g. AWS OpenSearch Ingestion Service /
OpenSearch Data Prepper's `otel_trace_source`) reject the **entire** OTLP
export request when any event in the batch has an empty name:

```
org.opensearch.dataprepper.model.trace.DefaultSpanEvent:
checkArgument(!builder.name.isEmpty(), "name cannot be an empty string");
```
https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-api/src/main/java/org/opensearch/dataprepper/model/trace/DefaultSpanEvent.java

Because Data Prepper's `OTelTraceGrpcService` rejects the whole gRPC
request on this validation error, a single OpenTracing `LogFields` call
anywhere in a batch causes the OTel collector to receive an HTTP 400 for
the entire export, marks it as a `Permanent`/non-retryable error, and drops
every span in that batch — not just the one with the empty-name event.

### Real-world trigger

Observed via Thanos (`thanos-io/thanos`), which vendors Cortex's cache
instrumentation and calls the OpenTracing API directly:

```go
// internal/cortex/chunk/cache/instrumented.go
sp := ot.SpanFromContext(ctx)
sp.LogFields(otlog.Int("keys requested", len(keys)))
...
sp.LogFields(otlog.Int("keys found", len(found)), otlog.Int("keys missing", len(keys)-len(found)))
```

Thanos wires this through `go.opentelemetry.io/otel/bridge/opentracing`
(`pkg/tracing/migration/bridge.go`), so every one of these `LogFields` calls
becomes an OTel span event with `name: ""`, ultimately causing OTel
collectors exporting Thanos traces to an OSIS/Data Prepper endpoint to have
their entire batch rejected with HTTP 400 and silently dropped.

### What I expected to happen

`LogFields`/`LogKV` should produce a non-empty OTel event name — e.g. use
the value of a field keyed `"event"` per OpenTracing convention, and fall
back to some constant (e.g. `"log"`) when no such field is present, instead
of unconditionally passing `""`.

### What happened instead

The event name is always `""`, which is spec-non-compliant per
`opentelemetry-proto`'s own documentation of `Event.name`, and causes silent,
whole-batch trace data loss against any backend that validates the field
(e.g. AWS OSIS / OpenSearch Data Prepper).

### Environment

- `go.opentelemetry.io/otel/bridge/opentracing` v1.36.0
- Observed indirectly via `thanos-io/thanos` (Cortex-derived cache
  instrumentation) exporting to AWS OpenSearch Ingestion Service
  (`opensearch-project/data-prepper`'s `otel_trace_source`)

### Additional context

Repro: ran an unmasked OSS Data Prepper instance (`otel_trace_source` ->
`stdout` sink) against captured OTLP export traffic and confirmed the
raw error:

```
Failed to parse request with error 'name cannot be an empty string'.
```

with every rejected span's `events{}` blocks having attributes (e.g.
`"keys requested"`, `"keys found"`) but no `name` field set at all.

### PR DESCRIPTION
**Pull request by singhlovepreet9**
Fixes #8643

## Problem

`bridgeSpan.LogFields` and `bridgeSpan.logRecord` translate every OpenTracing `Log`/`LogFields`/`LogKV`/`LogEvent` call into an OTel span event with a hardcoded empty name:

```go
s.otelSpan.AddEvent(
	"",
	trace.WithAttributes(otLogFieldsToOTelAttrs(fields)...),
)
```

`Span.Event.name` in `opentelemetry-proto` is documented as "semantically required to be set to non-empty string", so backends that validate it (e.g. AWS OSIS / OpenSearch Data Prepper) reject the **entire** OTLP export request as a permanent error, silently dropping every span in the batch — not just the offending one.

## Root cause

The bridge never implemented the event-name rule from the [OpenTracing compatibility specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/compatibility/opentracing.md#log):

> The `Add Event`'s `name` parameter MUST be the value with the `event` key in the pair set, or else fallback to use the `log` literal string.

## Fix

Derive the event name from the converted log fields — the value of the `event` field, falling back to the `log` literal — and pass it to `AddEvent`. Attributes are unchanged: the `event` field is still emitted as an attribute, matching the reference implementation in `opentelemetry-java`'s `SpanShim`.

Two deliberate details:

- When the same field set carries more than one `event` key, the last one wins, consistent with the last-value-wins duplicate handling used elsewhere in this repository.
- An `event` field with an empty value is treated as absent and falls back to `log`, so the bridge can never emit the empty name that causes the reported data loss. (The Java shim would propagate the empty string here; producing a non-empty name seemed strictly better given the failure mode this fixes.)

Out of scope: the specification's `event=error` → exception-semantics mapping (`RecordException`, `error.kind`/`message`/`stack` → `exception.*`) is a separate gap in this bridge and is not addressed here.

## Verification

New table-driven regression test `TestBridgeSpan_LogEventName` covers `LogFields`, `LogKV`, `LogEvent`, `LogEventWithPayload`, `Log`, and `FinishWithOptions` log records, plus the edge cases: no `event` field, empty `event` value, non-string `event` value, lazily emitted `event` field, and duplicate `event` fields.

Confirmed it reproduces the bug — with the change to `bridge.go` stashed, all 13 subtests fail (`expected: "cache.miss"` / `actual: ""`, `expected: "log"` / `actual: ""`).

```
$ cd bridge/opentracing && go test -run '^TestBridgeSpan_LogEventName$' ./...
ok  	go.opentelemetry.io/otel/bridge/opentracing	2.152s

$ go test -race -count=2 ./...
ok  	go.opentelemetry.io/otel/bridge/opentracing	2.351s
?   	go.opentelemetry.io/otel/bridge/opentracing/migration	[no test files]

$ make golangci-lint/bridge/opentracing
golangci-lint bridge/opentracing
0 issues.

$ gofmt -l bridge/opentracing/   # no output
```

`make precommit` in full was not run locally (it builds every module in the repo and this machine ran out of disk); the module's own build, vet, tests, gofmt and golangci-lint are all clean, and no dependencies changed.

### CODE REVIEW

**Comment by NesterovYehor** [comment]
LGTM

**Comment by MrAlias** [comment]
@singhlovepreet9 are you able to sign the CLA? If not we need to close this.

**Comment by singhlovepreet9** [comment]
@MrAlias Thanks for pointing this out! I have updated the commit to use the email address associated with my GitHub account which is covered by the CLA. The EasyCLA check should pass now.

**Comment by singhlovepreet9** [comment]
I've just rebased the PR against the latest `main` branch to resolve the merge conflict in `CHANGELOG.md`. This is now merge-ready!

File: bridge/opentracing/bridge_test.go [hunk 1]
```diff
@@ -896,6 +896,104 @@ func TestBridgeSpan_LogFields(t *testing.T) {
...
+		{
+			name: "LogFields with a lazily emitted event field",
+			log: func(s *bridgeSpan) {
+				s.LogFields(otlog.Lazy(func(e otlog.Encoder) { e.EmitString("event", "lazy") }))
+			},
+			expected: "lazy",
+		},
+		{
+			name:     "LogFields with duplicate event fields",
+			log:      func(s *bridgeSpan) { s.LogFields(otlog.Event("first"), otlog.Event("last")) },
```
Comment (ps-mir) [inline bridge/opentracing/bridge_test.go:934]: Nit: Consider adding a test that skips empty event
```
LogFields(otlog.Event("first"), otlog.Event(""))
```

returns `first`

**Review by ps-mir — APPROVED** [review]

File: bridge/opentracing/bridge.go [hunk 2]
```diff
@@ -224,6 +227,30 @@ func otLogFieldsToOTelAttrs(fields []otlog.Field) []attribute.KeyValue {
...
+// defaultEventName. An event name is semantically required to be non-empty, so
+// an empty value is treated as absent.
+func otelEventName(attrs []attribute.KeyValue) string {
+	for _, attr := range slices.Backward(attrs) {
+		if attr.Key != eventNameKey {
+			continue
+		}
+		if name := attr.Value.String(); name != "" {
+			return name
+		}
```
Comment (MrAlias) [inline bridge/opentracing/bridge.go:249]: When the last `event` field is empty, this continues scanning and returns an earlier non-empty duplicate. `LogFields(otlog.Event("first"), otlog.Event(""))` therefore records `"first"`, even though this PR defines the last field as authoritative and empty as falling back to `"log"`.

Could we stop at the first matching key during the reverse scan and add this combined case to `TestBridgeSpan_LogEventName`?

```suggestion
		if name := attr.Value.String(); name != "" {
			return name
		}
		return defaultEventName
```

**Review by MrAlias — COMMENTED** [review]
The event-name mapping covers the OpenTracing log paths, but the last-value-wins fallback still has one edge case to correct.

File: bridge/opentracing/bridge.go (hunk 2 above)
Comment (singhlovepreet9) [reply bridge/opentracing/bridge.go:249]: Good catch! Fixed — the reverse scan now stops at the first matching `eventNameKey` and returns `defaultEventName` when the value is empty, instead of continuing to scan for earlier non-empty duplicates. Added the `"LogFields with duplicate event fields last empty"` test case to cover this edge case. All tests pass.

**Review by MrAlias — APPROVED** [review]

**Review by dashpole — APPROVED** [review]
